### ISSUE TITLE
[BUG] Cannot Create a Service for Router nodes using the gRPC API

### ISSUE THREAD

**Issue by pjrwlazlo** [coreemu/core#595]
**Describe the bug**
Cannot create a custom services using the gRPC API functions. I have tried to make a new service for our filewall nodes that will automatically load the iptable rules I have generated, but it seems that I can only change services that are already enabled in CORE. I cannot add services to CORE as the gRPC documentation indicates under section "Configuring A Service" : http://coreemu.github.io/core/grpc.html 

**To Reproduce**
First we find all the routers in CORE, then we use the following to update IPForward with a new iptable rule

            `print("Found me a router")
            firewall_id = G.nodes[node]['core_node_id']
            # configure a service, for a node, for a given session
            core.set_node_service(
                session_id,
                firewall_id,
                "IPForward",
                files=["ipforward.sh"],
                directories=["/etc/node"],
                startup=["bash ipforward"],
                validate=[],
                shutdown=[],
            )
            # to edit the contents of a generated file you can specify
            # the service, the file name, and its contents
            core.set_node_service_file(
                session_id,
                firewall_id,
                "IPForward",
                "ipforward.sh",
                "iptables -A INPUT -p tcp -s 10.10.10.10 -j DROP",
            )`

**Expected behavior**
I expect to be able to create a new services such as "IPTables" and load the iptable rules I have into that new services, but I cannot create a new service. I can only edit old services.

**Screenshots**
As you can see there is a new empty file directory created from the code above, but the IPFoward file in the base directory is updated with our new iptable rule. When I try to create the new "IPTables" service, nothing is shown and the iptables rules are not added to any of the iptable chains with I use the command "iptables -L".

![image](https://user-images.githubusercontent.com/9749246/130711069-45cb80a3-d50e-49f0-8f4e-ca94e0b742be.png)
![image](https://user-images.githubusercontent.com/9749246/130711173-97d781da-2b36-4548-b76b-f8b0f40ca0ba.png)

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04

**Additional context**

CORE daemon v.6.5.0 instantiates Linux network namespace nodes.

**Comment by bharnden** [comment]
That explains how gRPC can be used to configure existing services, not for creating new ones. If you want to create a new service, you could do so, as noted [here](https://coreemu.github.io/core/services.html#new-services).

**Comment by bharnden** [comment]
You could write a custom service to create the files you need. They would be created relative to the nodes path, and available when you run something within a node.

Since nodes share the host file system, the default path for a node would be something like:
`/tmp/pycore.<session id>/<node name>.conf`

But, if you have pre defined files you would like to leverage, specifying absolute paths to these files is probably best. You can place them somewhere consistently, which could be a constant across machines even (outside a user path).

**Comment by pjrwlazlo** [comment]
So here is how we configure the services for the routing node in core:

`routerNode = Node(type=NodeType.Default, postion=its_position, model = 'router', name=its_name, services=['Zebra', 'StaticRoute', 'IPForward', 'Firewall'] )`

In the `core-pygui` we are not able to access the information, and only see a bash script:

![image](https://user-images.githubusercontent.com/9749246/131197919-63c755a0-6946-45ab-b3ff-c44ca6f2729f.png)

Here is using the command `vcmd -c /tmp/pycore.<session id>/<node name>` to access the terminal and it still shows us a bash screen. With bash, we cannot use vi, gedit, or any of tools on the host OS.

![image](https://user-images.githubusercontent.com/9749246/131197579-c8bc14b0-e836-4f35-a791-6111280b4810.png)

**Comment by pjrwlazlo** [comment]
@bharnden, I was wonder if you had any suggestion on how to solve this? I have been having this issue every since I added my custom "Firewall" service to the router nodes through gRPC. I can access bash just fine but vi, gedit, vim, will not work in bash.

**Comment by bharnden** [comment]
You can use anything that is on PATH for that terminal. `vi` may not, but likely `vim` would be.  I am not sure on your exact system setup, but have not seen anyone have similar issues as yourself. Typically you would be the root user.

Are you saying if you do not use the Firewall service, it works fine? Then I would say it has something to do with whatever your service is doing.

**Comment by bharnden** [comment]
I have never heard of anything like this and without your actual system to debug, it's really hard to say.

If you create a node through the GUI and create a basic "PC" node, start it, observe behavior and then stop.

Then add your new custom service, and repeat, do you see the same thing?

**Comment by pjrwlazlo** [comment]
436f756c6420796f7520706c6561736520656d61696c206d6520617420706a72776c617a6c6f40676d61696c2e636f6d2c20492063616e206578706c61696e206974206265747465722074686572652e204920776f756c64206a757374206e656564203330206d696e20746f20616e20686f7572206f6620796f75722074696d652e

**Comment by pjrwlazlo** [comment]
I think we found out the problem, we were adding iptable rules to the Firewall file and the firewall file does not have the iptables application listed under the `executables: Tuple[str, ...] = ()`. This seems to have broke some the services running on CORE.

1. We ended up creating our own service using this documentation: [myService in CORE](https://coreemu.github.io/core/services.html#example-custom-service )
2. Adding the link in the core-daemon for like `custom_services_dir = /home/username/.core/myservices`
3. Then adding 'iptables' in the executables to make: `iptables executables: Tuple[str, ...] = ("iptables",)`
Then the terminal window and other services  came back and the iptables loaded correctly.

Thank you very much for all your help @bharnden! 

I apologize I was not able to be more specific, some of this is unpublished work and I cannot disclose specific details until after it has been published.
